_posts/2016-11-22-files-cpp.md

```diff
@@ -0,0 +1,162 @@
+- [GCC support](https://gcc.gnu.org/projects/cxx-status.html#cxx11)
+- [Clang support](http://clang.llvm.org/cxx_status.html)
+- [Visual Studio support](https://msdn.microsoft.com/en-us/library/hh567368.aspx)
+
+## Third-party dependencies
+
+All of our third-party dependencies have *supported* building with a
+C++11 or C++14 compiler for some time.  However, more recently some of
+our dependencies have begin to *require* it, or will require it in
+their next major release.
+
+- Qt 5.7 (released June 2016) [requires
+  C++11](http://blog.qt.io/blog/2016/06/16/qt-5-7-released/)
+- ICU 59 will (expected April 2017) is [expected to require
+  C++11](https://sourceforge.net/p/icu/mailman/message/35232691/)
+  (delayed from ICU 58 to 59)
+- Other libraries we are considering using require C++11
+
+Once these libraries are adopted by Linux distributions, MacOS X
+homebrew, FreeBSD ports and other packaging systems, OME Files C++
+will no longer build on these systems.  While some of these are
+optional (Qt5), others such as ICU are required by Boost and
+Xerces-C++ for full Unicode support, which is needed by the OME-XML
+data model.  As a result, C++11 support will become a necessity for
+continued use of these libraries as the versions requiring C++11 enter
+mainstream use.
+
+## The benefits of C++11
+
+C++11 adds a number of useful features to the language, including:
+
+- `auto` type
+- `decltype` keyword
+- `default`ed and `delete`d functions
+- delegating constructors
+- forward-declared `enum`s
+- initialiser lists
+- lambdas
+- `nullptr` keyword
+- `override` and `final` qualifiers
+- range-based `for` loops
+- rvalue references
+- `static_assert`
+- strongly-typed `enum`s
+- variadic templates
+
+and library features, including:
+
+- `std::array`
+- `std::shared_ptr`
+- `std::thread`
+- `std::tuple`
+- `std::unique_ptr`
+
+These features are all supported by the compilers on the above
+platforms.  Some additional features (not included here) are not yet
+supported by the compilers on every platform, and so can't yet be
+used.
+
+These features benefit the project in several ways:
+
+- they reduce the code size dramatically by permitting much more
+  concise code with reduced boilerplate (by around 50% in one test
+  conversion of a single component)
+- they allow the use of standard library features in place of Boost
+  equivalents, which aids interoperability (e.g. `std::thread` and
+  `std::shared_ptr` in place of `boost::thread` and
+  `boost::shared_ptr`)
+- they allow improved performance of standard library containers
+  through the use of rvalue references, and also allow the use of new
+  features such as `std::unique_ptr` to replace `boost::shared_ptr` in
+  certain situations, again improving performance
+- they make the library easier to use, for example the use of
+  initialiser lists can reduce the `PixelBuffer` nD array addressing
+  from multiple lines to a single line of code
```

Comment:
I think one thing that you can also say here, admittedly it is mainly reflective of Windows C++ developers, is that it also makes the library easier to understand. Especially for people who have been exposed to these features for some time and expect an API to utilise them.

As you've already said, while Visual Studio does often lag behind in C++ standard feature thoroughness it has absolutely been the historical leader in selected feature availability. The regular Visual Studio releases obviously help this. As a consequence many Windows C++ developers are likely to be somewhat unfamiliar with what they might consider to be "antiquated" ways of exposing a C++ API.